qa/tasks/cephfs/filesystem.py

```diff
@@ -140,6 +140,31 @@ def get_mds_addr(self, name):
             log.warn(json.dumps(list(self.get_all()), indent=2))  # dump for debugging
             raise RuntimeError("MDS id '{0}' not found in map".format(name))
 
+    def cmpincname(self, inc, fsname):
+        """
+        Check the fs_name and incarnation values.
+        """
+        for fs in self.map['filesystems']:
+            if fs['mdsmap']['fs_name'] == fsname:
+                for info in fs['mdsmap']['info'].values():
+                    if info['incarnation'] == inc:
+                        return True
+                    else:
+                        return False
+        return False
+
+    def hadfailover(self, status):
+        """
+        Compares two statuses for mds failovers.
+        Returns True if the statuses are matching.
+        """
+        for fs in status.map['filesystems']:
+            for info in fs['mdsmap']['info'].values():
+                if not self.cmpincname(info['incarnation'], fs['mdsmap']['fs_name']):
```

Comment:
At this point you have the MDS info which includes a unique `gid`. Let's use that so you can look it up.

````
def get_mds_gid(self, gid):
    # like get_mds() but with gid
````

Then you can do in this code:

````
oldinfo = self.get_mds_gid(info['gid'])
if oldinfo is None or oldinfo['incarnation'] != info['incarnation']:
...
````

and then drop the `cmpincname` method.

Reply:
That's better way of doing it! I saw that code earlier, but didn't thought doing that way. Thanks.